salt/_modules/metalk8s_kubernetes_utils.py

```diff
@@ -184,3 +184,58 @@ def get_service_endpoints(service, namespace, kubeconfig):
         raise CommandExecutionError(error_tpl.format(service, namespace)) from exc
 
     return result
+
+
+def get_service_ips_and_ports(service, namespace, **kwargs):
+    try:
+        service_object = __salt__["metalk8s_kubernetes.get_object"](
+            name=service,
+            kind="Service",
+            apiVersion="v1",
+            namespace=namespace,
+            kwargs=kwargs,
+        )
+        if not service_object:
+            raise CommandExecutionError("Service not found")
+    except CommandExecutionError as exc:
+        raise CommandExecutionError(
+            f"Unable to get kubernetes service {service} in namespace {namespace}"
+        ) from exc
+
+    ips = service_object["spec"].get("clusterIPs")
+    if not ips:
+        ip = service_object["spec"].get("clusterIP")
+        if ip:
+            ips = [ip]
```

Comment:
I prefer the other way around, first take the `clusterIP` and if it's `None` or empty, use the `clusterIPs` first ip (I don't think it's possible to have `clusterIP` empty and `clusterIPs` list not empty but I may be wrong so let's keep this logic) 

And always return a single IP instead of a list of IPs

Reply:
I propose this solution :
```
    cip = service_object["spec"].get("clusterIP")
    cips = [cip] if cip else []
    cips.extend([ip for ip in service_object["spec"].get("clusterIPs", []) if ip != cip])
`
``

It will always return a list having the `clusterIP` as first element if it is present.